### PR TITLE
Document proto 1.5.0

### PR DESCRIPTION
**Pull request by TheRealPear**
* Changed recommended proto to 1.5.0
* Added a section in proto page covering filter & variables IDs, non-match scoped triggers excluding observers, and action untrigger.
* Replaced all mentions of 1.4.2 with 1.5.0 in examples where it was relevant.

Let me know if I've missed anything or should clarify more on something 👍

### CODE REVIEW

File: docs/modules/general/proto.mdx [hunk 1]
```diff
@@ -42,10 +42,40 @@ Mapmakers should always use the latest supported proto version, and this may be
...
+  - This applies to the following variables: `lives`, `score`, `timelimit`, `maxbuildheight`, `player.x`, `player.y`, `player.z`, `player.pitch`, `player.yaw`, `player.dir_x`, `player.dir_y`, `player.dir_z`, `player.vel_x`, `player.vel_y`, `player.vel_z`, `player.has_target`, `player.target_x`, `player.target_y`, `player.target_z`, `player.place_x`, `player.place_y`, and `player.place_z`.
+- Triggers and switch-scope actions now exclude observers when the scope is `team` or `player`.
+  - For maps below proto 1.4.2, `observers` defaults to `true` for backward compatibility. In proto 1.5.0, it defaults to `false`.
+
+    ```xml
+    <!-- Sample proto 1.4.2 XML -->
+    <trigger scope="player" observers="true" filter="your-filter">
+    <switch-scope inner="player" observers="true">
+    ```
+- Actions now untrigger by default.
```
[the remark this answers is not in the document]
Comment (TheRealPear) [reply docs/modules/general/proto.mdx:70]: This is more of a Pablo question, I only rewrote what he posted in PGMDev/PGM#1379

[the remark this answers is not in the document]
Comment (Pablete1234) [reply docs/modules/general/proto.mdx:70]: > I'm a bit lost. Didn't they always untrigger the moment a filter stops matching?

Nope, first version of actions wasn't propagating the untrigger (consider it a bug, idk), but to keep backwards-compat i made the untrigger-filter default to never.
This is corrected with proto 1.5 to be the behavior that makes the most sense

File: docs/modules/general/proto.mdx [hunk 2]
```diff
@@ -42,10 +42,40 @@ Mapmakers should always use the latest supported proto version, and this may be
...
 :::note
-`1.5.0` is a upcoming protocol version that is currently in its planning stage and therefore, **not implemented**.
-If you would like to learn more or propose changes, please visit [PGM#1267 on the issue tracker](https://github.com/PGMDev/PGM/issues/1267).
+`1.5.0` is now available. As the latest protocol version, it is not fully polished, so bugs are expected.
+If you would like to learn more or report issues, please visit the [PGM issue tracker](https://github.com/PGMDev/PGM/issues).
 :::
 
+#### Breaking
+
+- Filters and variables are now their own IDs that will be reserved by PGM. Those IDs can be referenced anywhere in the XML.
```
Comment (Pablete1234) [inline docs/modules/general/proto.mdx:51]: This wording makes it sound like all filters/variables behave like this; that is not the case.
This is only true for some filters & variables (the ones that essentially took no extra parameters), note that this isn't a new behavior, things like `never` or `everywhere` did exactly the same in the past, there's just now a bunch more reserved ids for commonly-used filters.

File: docs/modules/general/proto.mdx [hunk 3]
```diff
@@ -42,10 +42,40 @@ Mapmakers should always use the latest supported proto version, and this may be
...
+  - This applies to the following filters: `observing`, `participating`, `alive`, `dead`, `match-idle`, `match-starting`, `match-running`, `match-finished`, `match-started`, `crouching`, `walking`, `sprinting`, `grounded`, `flying`, `can-fly`, and `void`.
+  
+    ```xml
+    <!-- Sample proto 1.4.2 XML -->
+    <filters>
+        <alive id="alive"/> <!-- id="alive" is not necessary in proto 1.5.0 -->
+    </filters>
```
Comment (Pablete1234) [inline docs/modules/general/proto.mdx:58]:
```suggestion
    <!-- Sample proto 1.4.2 XML -->
    <filters>
        <alive id="alive"/> 
    </filters>
    <some-other-feature filter="alive"/>
    
    <!-- Sample proto 1.5.0 XML -->
    <filters>
      <!-- Defining the alive id is not necessary, as it's a built-in now -->
    </filters>
    <some-other-feature filter="alive"/>    
```
What's not necessary isn't the `id="alive"`, but rather the whole defining of that filter in the first place, in order for it to be usable in other features

File: docs/modules/general/proto.mdx (hunk 2 above)
Comment (TheRealPear) [reply docs/modules/general/proto.mdx:51]: I completely forgot about `never` and `everywhere`. Were there any variables with this behavior before proto 1.5.0? If not, would this be a good change?

```suggestion
- Variables and some additional filters are now their own IDs that will be reserved by PGM. Those IDs can be referenced anywhere in the XML.
```

Comment (Pablete1234) [reply docs/modules/general/proto.mdx:51]: Variables were all added at modern times, the latest proto (1.4.2) pre-dates them by years, so no we could not have reserved any id for them.
The updated text looks better, but it still sounds like it applies to all variables, maybe `Some variables and some additional filters`?

**Review by CoWinkKeyDinkInc — APPROVED** [review]

**Pull request merged**